app/com/lynxanalytics/biggraph/frontend_operations/WorkflowOperations.scala

```diff
@@ -501,6 +501,83 @@ class WorkflowOperations(env: SparkFreeEnvironment) extends ProjectOperations(en
     registerSQLOp(s"SQL$inputs", numbers.take(inputs))
   }
 
+  registerOp("Filter with SQL", "filter", category, List("input"), List("output"), new SmartOperation(_) {
+    private val input = context.inputs("input")
+    if (input.isProject) {
+      params ++= List(
+        Code("vertex_filter", "Vertex filter", language = "sql"),
+        Code("edge_filter", "Edge filter", language = "sql"))
+    } else {
+      params ++= List(
+        Code("filter", "Filter", language = "sql"))
+    }
+    override def summary = {
+      if (input.isProject) {
+        val vf = params("vertex_filter").trim
+        val ef = params("edge_filter").trim
+        if (vf.isEmpty && ef.isEmpty) "Filter with SQL"
+        else if (vf.nonEmpty && ef.nonEmpty) s"Filter to $vf; $ef"
+        else s"Filter to $vf$ef"
+      } else {
+        val f = params("filter").trim
+        if (f.isEmpty) "Filter with SQL"
+        else s"Filter to $f"
+      }
+    }
+    def enabled = FEStatus.enabled
+    override def getOutputs() = {
+      params.validate()
+      if (input.isProject) {
+        val project = input.project
+        def tables = project.viewer.getProtoTables.toMap
+        val before = project.rootEditor.viewer
+        if (params("vertex_filter").trim.nonEmpty) {
+          project.vertexAttributes.updateEntityMap(
+            project.vertexAttributes.iterator.toMap + ("!id" -> project.vertexSet.idAttribute))
+          val vf = graph_operations.ExecuteSQL.run(
+            "select `!id` from vertices where " + params("vertex_filter"), tables)
+          project.vertexAttributes.updateEntityMap(
```

Comment:
> This back and forth editing of the project is a bit hacky. Why not create a new copy of the project for the generation of the table instead of editing "the" project?

Done.